### PR TITLE
remove some duplicate dependencies

### PR DESCRIPTION
**Pull request by pjgg**
### Summary

- Remove some duplicated dependencies

Please select the relevant options.

- [X] Refactoring

### Checklist:
- [X] Commits are well encapsulated and follow [the best practices](https://cbea.ms/git-commit/)

### CODE REVIEW

File: messaging/kafkaSSL/pom.xml [hunk 1]
```diff
@@ -31,11 +31,6 @@
             <artifactId>kafka</artifactId>
             <scope>test</scope>
         </dependency>
-        <dependency>
```
Comment (jsmrcka) [inline messaging/kafkaSSL/pom.xml:34]: There is one more occurence at https://github.com/quarkus-qe/quarkus-test-suite/blob/main/messaging/infinispan-grpc-kafka/pom.xml#L46.

File: messaging/kafka-producer/pom.xml [hunk 2]
```diff
@@ -18,16 +18,6 @@
...
-            <scope>test</scope>
-        </dependency>
-        <dependency>
-            <groupId>io.rest-assured</groupId>
-            <artifactId>rest-assured</artifactId>
-            <scope>test</scope>
-        </dependency>
         <dependency>
             <groupId>io.quarkus.qe</groupId>
             <artifactId>quarkus-test-containers</artifactId>
```
Comment (jsmrcka) [inline messaging/kafka-producer/pom.xml:23]: This one is in the parent too, so I think we can remove it from here.

File: service-binding/postgresql-crunchy/pom.xml [hunk 3]
```diff
@@ -39,16 +39,6 @@
...
-            <scope>test</scope>
-        </dependency>
-        <dependency>
-            <groupId>io.rest-assured</groupId>
-            <artifactId>rest-assured</artifactId>
-            <scope>test</scope>
-        </dependency>
         <dependency>
             <groupId>io.quarkus.qe</groupId>
             <artifactId>quarkus-test-openshift</artifactId>
```
Comment (jsmrcka) [inline service-binding/postgresql-crunchy/pom.xml:44]: This one is in the parent too, so I think we can remove it from here.

**Review by jsmrcka — CHANGES_REQUESTED** [review]

File: service-binding/postgresql-crunchy/pom.xml (hunk 3 above)
Comment (jsmrcka) [reply service-binding/postgresql-crunchy/pom.xml:44]: I still see `quarkus-test-openshift` here, is it necessary then, or can we remove it?

**Review by jsmrcka — CHANGES_REQUESTED** [review]

File: service-binding/postgresql-crunchy/pom.xml (hunk 3 above)
Comment (pjgg) [reply service-binding/postgresql-crunchy/pom.xml]: done!

**Review by jsmrcka — APPROVED** [review]

**Comment by pjgg** [comment]
Serverless issues on CI are not related to this PR

**Pull request merged**